src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js

```diff
@@ -259,21 +322,33 @@ export class RuleEditor extends LinkedComponent {
         this.props.t('rules.flyouts.ruleEditor.validation.required')
       );
     this.conditionsLink = this.formDataLink.forkTo('conditions').withValidator(requiredValidator);
+    this.actionsLink = this.formDataLink.forkTo('actions').withValidator(requiredValidator);
     this.severityLink = this.formDataLink.forkTo('severity');
-    //todo toggle button didn't support link
-    this.enabledLink = this.formDataLink.forkTo('enabled');
+
     // Create the state link for the dynamic form elements
     const conditionLinks = this.conditionsLink.getLinkedChildren(conditionLink => {
       const fieldLink = conditionLink.forkTo('field').map(({ value }) => value).withValidator(requiredValidator);
       const operatorLink = conditionLink.forkTo('operator').map(({ value }) => value).withValidator(requiredValidator);
       const valueLink = conditionLink.forkTo('value')
-        .check(Validator.notEmpty, () => this.props.t('deviceGroupsFlyout.errorMsg.nameCantBeEmpty'))
+        .check(Validator.notEmpty, this.props.t('deviceGroupsFlyout.errorMsg.isRequired'))
         .check(val => !isNaN(val), t('rules.flyouts.ruleEditor.validation.nan'));
       const error = fieldLink.error || operatorLink.error || valueLink.error;
       return { fieldLink, operatorLink, valueLink, error };
     });
 
+    const actionLinks = this.actionsLink.getLinkedChildren(actionLink => {
+      const emailLink = actionLink.forkTo('parameters')
```

Comment:
It would be better to save the parameters link and reuse it for email and template. forkTo creates a new object so might as well reuse it.
```
const parametersLink = actionLink.forkTo('parameters');
const emailLink = parametersLink.forkTo('email')
  .check(Validator.notEmpty, this.props.t('deviceGroupsFlyout.errorMsg.isRequired'));
```

Reply:
Done